[IMP] payment_acquirers/stripe: split authorization and capture (#1642)
odoo/documentation: content/applications/finance/payment_acquirers/stripe.rst

When you click on **Add endpoint**, your Webhook is configured. You can then click on **reveal** to
display your signing secret.

Configuration tab
-----------------

> .. _stripe/place_hold_on_card:

Reviewer: `manual_capture` is fine. Let's not sacrifice all our technical tools to Google's gods.

Place a hold on a card
~~~~~~~~~~~~~~~~~~~~~~

With Stripe, you can enable the :ref:`manual capture <payment_acquirers/capture_amount>`. If
enabled, the funds are reserved for 7 days on the customer's card, but not charged yet.

.. warning::

>    Only some payments methods support separate authorization and capture.
>    
>    For example, card payments support separating these steps. The expiration window is 7 days.
> 
>    With payment methods that don’t support this functionality, like iDEAL, you can’t separate
>    the authorization and capture steps.

Reviewer 2: I wouldn't talk about other payment methods, I don't think it's useful since we're configuring stripe, I think the only important information here is the 7-day-window. However, I wouldn't write it like that, it looks like it's not important

Also I tried to rephrase it but I'm not convinced about what I came up with, is it the same as the Adyen 7-day-window? 
[https://www.odoo.com/documentation/master/applications/finance/payment_acquirers/adyen.html#place-a-hold-on-a-card](url)
```suggestion
    
   You have **7 days** to capture the amount. After this time, the manual capture expires.
```

Author: it suits me well :+1:

When you click on **Add endpoint**, your Webhook is configured. You can then click on **reveal** to
display your signing secret.

Configuration tab
-----------------

> .. _stripe/manual_capture:

Reviewer 3: I think we prefer using hyphens for internal references. Don't we @AntoineVDV ?
I know it isn't the case everywhere and that is not part of the guidelines.  It's written with hypens at line 134.

Reviewer: We usually do but it's not mandatory and it has no impact on the SEO AFAIK (the anchor is not based on the external hyperlink target). So there is no point in adding that to the guidelines. However, it's usually best to follow the page's styling when adding content to that page.

Author: Based on the styling of the other anchor on that page, I think I shouldn't have used hyphens at line 134 in my previous PR. It's the only anchor that uses hyphens on that page so I'll keep underscore here.

Place a hold on a card
~~~~~~~~~~~~~~~~~~~~~~

Stripe allows you to **capture an amount manually** instead of having an immediate capture. This
feature earmarks an amount on the customer's account for a determined period of time that depends on
the payment method used.

To set it up, enable the :guilabel:`Capture Amount Manually` option on Odoo, as explained in the
:ref:`payment acquirers documentation <payment_acquirers/capture_amount>`.

.. warning::

>    When the manual capture feature is activated, your customers won't be able to proceed with any
>    payment if unsupported payment methods are selected.

Reviewer 3: If you want to write future tense, then I'd replace "when" with "Once the manual capture, ..."
Otherwise, I'd rather go with the present tense since it's normal expected behavior.
```suggestion
   When the manual capture feature is activated, your customers are not able to proceed with any
   payment if unsupported payment methods are selected.
```

> .. caution::
>    Odoo doesn't support the partial capture yet. Be aware that if you make a partial capture from
>    Stripe's interface, Odoo will manage it as if it was a full capture.

Reviewer 2: ```suggestion
   .. caution::
   Odoo doesn't support partial capture yet. Be aware that a partial capture from Stripe's 
   interface will still be managed as a full capture.
```
Here's my suggestion for this part, I just rephrased it, but there was no problem with it to start with x)  
(just try to avoid the future tense, I couldn't manage to take it off x) )

Author: I adapted your rephrasing adding 'by Odoo' at the end because you CAN do a partial capture from Stripe, but in Odoo, it will show as if you capture all the amount, even if you only took half the amount.

When you click on **Add endpoint**, your Webhook is configured. You can then click on **reveal** to
display your signing secret.

Configuration tab
-----------------

.. _stripe/manual_capture:

Place a hold on a card
~~~~~~~~~~~~~~~~~~~~~~

Stripe allows you to **capture an amount manually** instead of having an immediate capture. This
feature earmarks an amount on the customer's account for a determined period of time that depends on
the payment method used.

To set it up, enable the **Capture Amount Manually** option on Odoo, as explained in the

> :ref:`payment acquirers documentation <payment_acquirers/capture_amount>`.

Reviewer 2: shouldn't we use " :guilabel:`Capture Amount Manually` "here?

> .. warning::
>    Some payment methods, unlike cards, don't support manual capture. When the manual capture feature 
>    is activated, your customers won't be able to proceed with their payment if unsupported payment 
>    methods are selected.

Reviewer 2: You might need to check the 100th character I didn't check it on PyCharm this time
```suggestion
.. warning::
When the manual capture feature is activated, your customers won't be able to proceed with any payment if unsupported payment methods are selected.
```

.. caution::

>    Odoo doesn't support the partial capture yet. Be aware that a partial capture from Stripe's 
>    interface is still be managed as a full capture by Odoo.

Reviewer 2: ```suggestion
   Odoo doesn't support the partial capture yet. Be aware that a partial capture from Stripe's 
   interface is still managed as a full capture by Odoo.
```
